### ISSUE TITLE
pdns check 4.0.3 fails to build with gcc 7

### ISSUE THREAD

**Issue by mortenstevens** [PowerDNS/pdns#5007]
Buildlog:

```
g++ -DHAVE_CONFIG_H -I. -I..  -I.. -I.. -pthread  -I../ext/json11 -I../ext/yahttp         -DLDAP_DEPRECATED -DSYSCONFDIR=\"/etc/pdns\" -DPKGLIBDIR=\"/usr/lib64/pdns\" -DLOCALSTATEDIR=\"/var/run\" -fPIE -DPIE -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 --param ssp-buffer-size=4 -fstack-protector -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic -c -o test-dnsname_cc.o test-dnsname_cc.cc
recpacketcache.cc: In member function 'uint64_t RecursorPacketCache::bytes()':
recpacketcache.cc:214:13: error: types may not be defined in a for-range-declaration [-fpermissive]
   for(const struct Entry& e :  d_packetCache) {
             ^~~~~~
make[3]: *** [Makefile:2529: recpacketcache.o] Error 1
make[3]: *** Waiting for unfinished jobs....
```

Full buildlog: https://kojipkgs.fedoraproject.org//work/tasks/9023/17769023/build.log

Do you have a patch available?

**Comment by Habbie** [comment]
`DEBUG util.py:435:   gcc-c++                      x86_64  7.0.1-0.7.fc26               build   11 M
`

**Comment by Habbie** [comment]
Can you try replacing `const struct Entry& e` with `const auto& e`?

**Comment by mortenstevens** [comment]
@Habbie 
Yes, I can try this patch: https://github.com/PowerDNS/pdns/commit/f226db2f2c12a2c0c16b3125a0438d9aca0d017c

**Comment by Habbie** [comment]
@mortenstevens ah, wonderful

**Comment by mortenstevens** [comment]
@Habbie 

Thank you. It's now building fine after applying commit f226db2f2c12a2c0c16b3125a0438d9aca0d017c
See: https://koji.fedoraproject.org/koji/buildinfo?buildID=859174

Fedora commit: https://src.fedoraproject.org/cgit/rpms/pdns.git/commit/?id=903f9360096ce9335f1a77888c26d464571bf0dd

**Comment by Habbie** [comment]
Fixed in #5029
